From: Matthew Flatt <mflatt@racket-lang.org>
Date: Mon, 11 Mar 2024 07:58:42 -0600
X-Dgit-Generated: 8.12+dfsg1-4 234f8d9ba0ee0efcff5e11a869fd892a5641200b
Subject: CS: repairs for detecting big-endian mode in boot file embedding

Fill out the list of big-endian platforms. I considered changes to
avoid depending on a list when not cross-compiling, but decided that
it is probably better to find out during a non-cross compile if the
list gets out-of-date.

Related to #4951

---

diff --git a/src/cs/c/embed-boot.rkt b/src/cs/c/embed-boot.rkt
index a9d531f44e..7b8860ed51 100644
--- a/src/cs/c/embed-boot.rkt
+++ b/src/cs/c/embed-boot.rkt
@@ -127,9 +127,16 @@
      (define terminator-len (bytes-length terminator))
 
      (define big-endian?
-       (if target
+       (if (and target
+                (not (equal? target (symbol->string (system-type 'target-machine)))))
            (case target
-             [("tppc32le") #t]
+             [("tppc32osx" "ppc32osx"
+                          "tppc32le" "ppc32le"
+                          "tppc32fb" "ppc32fb"
+                          "tppc32ob" "ppc32ob"
+                          "tppc32nb" "ppc32nb"
+                          "tpb64b" "pb64b"
+                          "tpb32b" "pb32b") #t]
              [else #f])
            (system-big-endian?)))
 
